jbpm-human-task/jbpm-human-task-audit/src/main/java/org/jbpm/services/task/audit/JPATaskLifeCycleEventListener.java

```diff
@@ -71,9 +75,10 @@ public void afterTaskStartedEvent(TaskEvent event) {
         Task ti = event.getTask();
         TaskPersistenceContext persistenceContext = getPersistenceContext(((TaskContext)event.getTaskContext()).getPersistenceContext());
         try {
-            TaskEventImpl taskEventImpl = new TaskEventImpl(ti.getId(), org.kie.internal.task.api.model.TaskEvent.TaskEventType.STARTED, ti.getTaskData().getProcessInstanceId(), ti.getTaskData().getWorkItemId(), userId);
+            TaskEventImpl taskEventImpl = new TaskEventImpl(ti.getId(), TaskEventType.STARTED, ti.getTaskData().getProcessInstanceId(), ti.getTaskData().getWorkItemId(), userId);
             event.getMetadata().put(METADATA_TASK_EVENT, taskEventImpl);
             persistenceContext.persist(taskEventImpl);
+            createView(event, TaskEventType.STARTED);
```

Comment:
I've got some doubts about how this 2-step approach impacts `EventEmitter` implementations which aim to be transactional and synchronous (e.g. sending the data to Elastic in `deliver()` instead of in `apply()`).

I understand that if `persist()` fails (e.g. connectivity error throwing a `RuntimeException` when sending the `TaskInstanceView` to Elastic), the operation is rolled-back, the process instance doesn't change state and `createView()` isn't called. That's the expected behavior.

Nevertheless, if the error happens in `createView()`, the first `TaskInstanceView` would have already been sent to Elastic, and the process instance would have already moved on. Nevertheless the `TaskOperationView` will never be stored in Elastic, hence the event information may be lost.

Is this correct?

Reply:
your understanding is correct, but thats the way the emitter works. It is executed in a trasnsactional context, but the emission itself is decoupled from that context

Reply:
Thanks @fjtirado . Indeed as you mentioned, I've tested that even if the 2 separate views are created, both are passed to the emitter's `deliver()` and `apply()` methods at the same time. Hence the emission is kept transactional (when using `deliver()` to send the data to Elasticsearch).